SpatialGDK/Source/SpatialGDKTests/SpatialGDKEditor/SpatialGDKEditorSchemaGenerator/SpatialGDKEditorSchemaGeneratorTest.cpp

```diff
@@ -1,9 +1,859 @@
+		{
+			"SpatialTypeActorWithMultipleObjectComponents",
+			ExpectedFileContent::ASpatialTypeActorWithMultipleObjectComponents
+		}
+};
+
+class SchemaValidator
+{
+public:
+	bool ValidateGeneratedSchemaForClass(const FString& FileContent, const UClass* CurrentClass)
+	{
+		if (FString* ExpectedContentPtr = ExpectedContents.Find(CurrentClass->GetName()))
+		{
+			FString ExpectedContent = *ExpectedContentPtr;
+			ExpectedContent.ReplaceInline(TEXT("{{id}}"), *FString::FromInt(GetID()));
+			return (FileContent.Compare(ExpectedContent) == 0);
+		}
+		else
+		{
+			return false;
+		}
+	}
+
+private:
+	int GetID()
+	{
+		return freeId++;
+	}
+
+	int freeId = 10000;
+};
+
+void DeleteTestFolders()
+{
+	IPlatformFile& PlatformFile = FPlatformFileManager::Get().GetPlatformFile();
+	PlatformFile.DeleteDirectoryRecursively(*FPaths::Combine(FPaths::ProjectContentDir(), TEXT("Spatial/Tests/")));
+	PlatformFile.DeleteDirectoryRecursively(*gSchemaOutputFolder);
+}
+
+}
+
+SCHEMA_GENERATOR_TEST(GIVEN_spatial_type_class_WHEN_checked_if_supported_THEN_is_supported)
+{
+	// GIVEN
+	const UClass* SupportedClass = USpatialTypeObjectStub::StaticClass();
+
+	// WHEN
+	bool bIsSupported = SpatialGDKEditor::Schema::IsSupportedClass(SupportedClass);
+
+	// THEN
+	TestTrue("Spatial type class is supported", bIsSupported);
+	return true;
+}
+
+SCHEMA_GENERATOR_TEST(GIVEN_class_derived_from_spatial_type_class_WHEN_checked_if_supported_THEN_is_supported)
+{
+	// GIVEN
+	const UClass* SupportedClass = UChildOfSpatialTypeObjectStub::StaticClass();
+
+	// WHEN
+	bool bIsSupported = SpatialGDKEditor::Schema::IsSupportedClass(SupportedClass);
+
+	// THEN
+	TestTrue("Child of a Spatial type class is supported", bIsSupported);
+	return true;
+}
+
+SCHEMA_GENERATOR_TEST(GIVEN_null_pointer_WHEN_checked_if_supported_THEN_is_not_supported)
+{
+	// GIVEN
+	const UClass* SupportedClass = nullptr;
+
+	// WHEN
+	bool bIsSupported = SpatialGDKEditor::Schema::IsSupportedClass(SupportedClass);
+
+	// THEN
+	TestFalse("Null pointer is not a valid argument", bIsSupported);
+	return true;
+}
+
+SCHEMA_GENERATOR_TEST(GIVEN_non_spatial_type_class_WHEN_checked_if_supported_THEN_is_not_supported)
+{
+	// GIVEN
+	const UClass* SupportedClass = UNotSpatialTypeObjectStub::StaticClass();
+
+	// WHEN
+	bool bIsSupported = SpatialGDKEditor::Schema::IsSupportedClass(SupportedClass);
+
+	// THEN
+	TestFalse("Non spatial type class is not supported", bIsSupported);
+	return true;
+}
+
+SCHEMA_GENERATOR_TEST(GIVEN_class_derived_from_non_spatial_type_class_WHEN_checked_if_supported_THEN_is_not_supported)
+{
+	// GIVEN
+	const UClass* SupportedClass = UChildOfNotSpatialTypeObjectStub::StaticClass();
+
+	// WHEN
+	bool bIsSupported = SpatialGDKEditor::Schema::IsSupportedClass(SupportedClass);
+
+	// THEN
+	TestFalse("Child of Non-Spatial type class is not supported", bIsSupported);
+	return true;
+}
+
+SCHEMA_GENERATOR_TEST(GIVEN_a_class_with_not_spatial_tag_WHEN_checked_if_supported_THEN_is_not_supported)
```

Comment:
`GIVEN_null_class_pointer_WHEN_checked_if_supported_THEN_is_not_supported`?

Reply:
yeah, why not? It's nice to know the function doesn't crash with nullptr